lmfit/minimizer.py

```diff
@@ -282,11 +290,16 @@ def __residual(self, fvars):
         if self._abort:
             return None
         params = self.result.params
-        for name, val in zip(self.result.var_names, fvars):
-            params[name].value = params[name].from_internal(val)
+
+        if apply_scaling:
+            for name, val in zip(self.result.var_names, fvars):
+                params[name].value = params[name].from_internal(val)
+                params.update_constraints()
+        else:
+            for name, val in zip(self.result.var_names, fvars):
+                params[name].value = val
```

Comment:
I wasn't sure if update_constraints needs the bound transformation to work, so thats why i left it out, but i am happy to put it back in.